### PR TITLE
(WIP) Loadout and Quirk update, now with more customization!

### PR DESCRIPTION
**Pull request by Ms-Mee**
## About The Pull Request

This PR changes the handling of loadout and quirks, lightly connecting the systems, making the respective UIs a bit more informative or otherwise nicer to navigate, and using a secondary window popup for a more detailed description of them, as well as for handling any secondary options they may have. Intended changes and fixes are better explained by the lists below.

### Quirk TODO/feature list:
- [X] Update the Quirk UI, make it more descriptive.
~~Add the in-depth quirk window. Make the in-depth quirk window prettier.~~ Quirk window was scrapped.
- [x] Fully implement customization for quirk values.
- [x] Add the customization options for applicable quirks.
- [ ] ~~Update the Family Heirloom quirk to instead take loadout items as heirlooms.~~ Mark items as emotional items in the Loaodout tab, giving quirk balance points and the quirk functionality without a quirk attached, if possible.
- [ ] Make it clearer that an item is one of the heirlooms of someone, and make crafting items with heirlooms update the crafted item to be a heirloom(example: crafting presciption sechud with prescription glasses marked as a heirloom).
- [X] ~~Find out why sometimes certain quirk instances just lose their assigned datum when the secondary options window closes and fix that, if the window closing is even the cause for some reason.~~ Get rid of the quirk instance list.
- [x] Add dynamic quirk values which vary based on how many options that affect them have been taken, currently intended only for handling multiple addictions, based on their severity, phobias and such. If possible add to prosthetics as well.
### Loadout TODO/feature list:
- [ ] Update Loadout UI.
- [ ] Implement taking multiple of certain loadout items.
- [ ] Update loadout spawn handling. Allow equipping items to slots and move default equipment to suitable storage. Add 2 special "packs" that contain leftover items, if they're not equipped anywhere and the user's backpack doesn't fit them. They can also contain objects such as wheelchairs, if needed, but nothing can ever be put inside them, only taken out.
- [X] Fix improper slot assignment for existing loadout items.
- [ ] Update existing loadout items to be compatible with new backend.
- [ ] Make it skip certain job items when applicable substitutes have been added, to avoid duplicating certain valuable items (example: hud prescription glasses replacing spawned prescription glasses and the default spawned hud glasses.)
- [ ] Rebalance loadout costs and limits.
### General TODO/feature list:
- [ ] Integrate Loadout and Quirks for applicable quirks (ie Nearsighted), making their spawned gear be handled by loadout, and making certain items avaialable through loadout where they otherwise weren't(instruments, possibly wheelchairs from paraplegic).
- [ ] Update changelog to include every change once the PR is fully finished.
- [ ] Comment and document changes to help anyone interacting with the systems understand how to work with them, mappers and coders alike. (might make an external explanation instead if requested)
- [ ] Add screenshots of updated/new UIs when they're complete.
- [ ] (EXTRA) Add config options for toggling certain features such as taking multiple of loadout items.
- [ ] (EXTRA) Add search bars for both, if possible, not sure about that.

Anything prefixed with (EXTRA) is not guaranteed to be implemented, simply something I plan on attempting or already started but might cut to a different PR if either requested or unviable within a reasonable timeframe when all the primary tasks have been finished.

- [ ] I affirm that I have tested all of my proposed changes and that any issues found during tested have been addressed.

## Why It's Good For The Game

It updates the loadout UI, partially by request of tira, makes the quirk UI more informative and increases customization for both quirks and loadout. Also increases mapper control over intended ship equipment, where necessary. UI gooder and more options = good, I think.

## Changelog

:cl:
add: Added quirk customization options, which allow selecting more than one given option for quirks such as junkie and phobia, also integrating selection to the quirk UI instead of having it left loose in the Character Appearance tab.
add: Added more things
fix: Fixed the black and white gloves not being shown in the character setup preview.
code: Changed the loadout gear spawn handling, changed the loadout and quirk UIs.
/:cl:

### CODE REVIEW

**Comment by Ms-Mee** [comment]
No more TGUI for now, I can explain if need be but it doesn't matter too much and is a bit too personal, I'll still try to be done with the PR before the 20th of february though I'm a little busy as is. If there is a need for a bit more clarification on what I'm doing and the current state of things, they haven't changed too much from when I showed off a few screenshots on the dev channel, though I'll send images here once I'm done with loadout to properly be able to address any possible problems and complaints with the new structure and elaborate on any unclear functions.

**Comment by Ms-Mee** [comment]
Little do they know just how horrifying the things of my creation are

**Comment by thgvr** [comment]
I am watching your career with great interest.

**Comment by Ms-Mee** [comment]
THE EYES

**Comment by Ms-Mee** [comment]
I could probably make a little recording showing the state of functionality and implementation of my plans so far now that I have a laptop that isn't a tween and an issue I was having sorta just fixed itself which allows me to test things again so I can get back to work, which I will probably later today

**Comment by Ms-Mee** [comment]
It's kinda hard to figure out how exactly I should and shouldn't implement some things especially with my lack of experience in UI design, but I do know how to get things _working_ and they mostly are

File: code/controllers/subsystem/processing/quirks.dm [hunk 1]
```diff
@@ -37,7 +45,37 @@ PROCESSING_SUBSYSTEM_DEF(quirks)
 		var/datum/quirk/T = V
 		quirks[initial(T.name)] = T
 		quirk_points[initial(T.name)] = initial(T.value)
-		quirk_instances[initial(T.name)] = new T
+		if(initial(T.additional_values))
```
Comment (Ms-Mee) [inline code/controllers/subsystem/processing/quirks.dm:48]: Honestly, what the fuck IS this?

**Comment by Ms-Mee** [comment]
I need to stop forgetting the good ideas I have while on the bus or at least be more decisive with how I do thiiings

**Comment by Ms-Mee** [comment]
I HATE MAPPERS
UPPER TEXT (I have been handed a minuscule inconvenience)

**Comment by Ms-Mee** [comment]
I wonder how too busy with calculus stuff to use my 4 "free" days to work on this I'll be, and how likely I am to get distracted by Zomboid when I can work on this

**Comment by ZephyrTFA** [comment]
I am closing this due to inactivity for over four months. When you are able to actively work on this please let a maintainer know.

**Pull request closed without merging**